src/EcomDestinationWriter.cs

```diff
@@ -715,6 +716,20 @@ private int LastDetailId
         set { _lastDetailId = value; }
     }
 
+    private int _lastStockLocationGroupId = -1;
+    private int LastStockLocationGroupId
+    {
+        get
+        {
+            if (_lastStockLocationGroupId == -1)
+            {
+                _lastStockLocationGroupId = GetLastId(CommandBuilder.Create("(select convert(nvarchar,MAX(CAST(StockLocationGroupId as int))) as lastID from EcomStockLocation)"));
+            }
+            return _lastStockLocationGroupId;
```

Comment:
If it gets set once, we rely on that number and just +1 it on line 1947, I'm not sure we can really rely on that, as the number could increase from other sources as well - I think a method that simply returns the next number is more suitable and use that the one place it needs to be used (line 1948)